core/src/main/scala/org/apache/spark/deploy/history/EventLogFileCompactor.scala

```diff
@@ -221,5 +221,5 @@ private class CompactedEventLogFileWriter(
     hadoopConf: Configuration)
   extends SingleEventLogFileWriter(appId, appAttemptId, logBaseDir, sparkConf, hadoopConf) {
 
-  override val logPath: String = originalFilePath.toUri.toString + EventLogFileWriter.COMPACTED
```

Comment:
Is this safe with S3 filesystem?

Reply:
This looks like to have unexpected regression, if we are not sure the change is compatible with other file systems.

Reply:
Ya, it looks like that to me. Especially with Apache Hadoop 3.3.2.

Reply:
Is it because the string presentation of the path is able to omit the scheme of the URI, or it does not encode the special characters to the encoded chars? Actually, we would have to always use Hadoop's `Path` to work on Hadoop's `Path` so `fs.default.name` is respected.

The only exception would be when this path is stored somewhere outside. In this case, the URI has to be stored as the fully-qualified one so different `fs.default.name` does not affect the original path.

This path URI stuff has many weird holes so we're just delegating to Hadoop's `Path` implementation, and leveraging their bug fixes.

I am fine if this change does not handle all of the cases but wanted to point out this direction is better than manually handling it in Spark.